pkgs/applications/version-management/sourcehut/todo.nix

```diff
@@ -30,9 +31,34 @@ buildPythonPackage rec {
     export SRHT_PATH=${srht}/${python.sitePackages}/srht
   '';
 
-  # Tests require a network connection
   doCheck = false;
 
+  checkInputs = [
+    pytest
+    factory_boy
+  ];
+
+  checkPhase = let
```

Comment:
Why do you add this if `doCheck` is still `false`?

Reply:
The python packaging doesn't care about `doCheck`, it does it's own `setuptoolsCheckPhase` which runs unless a `checkPhase` is declared.

Reply:
There is a separate boolean but I'm assuming those tests should be run since it's just testing the python code.

Reply:
If I'm not mistaken `checkPhase` is not supposed to work for python. Even if I'm wrong, we should probably try to clarify in comments what is going on here.

Reply:
https://github.com/NixOS/nixpkgs/blob/65b1948be2f9e3ad0cc7544f8727d6f3c3f49c9f/pkgs/development/interpreters/python/mk-python-derivation.nix#L100-L101

Reply:
The error isn't coming from the `installCheckPhase` and it seems they are just aliases to `doCheck` and `checkPhase`. Then depending on how the package is installed, other phases are added, in this specifically, the setuptools specific hooks [1].

[1] https://github.com/NixOS/nixpkgs/blob/731c82a447d916e78f31642ad757a3ce8911f1ea/pkgs/development/interpreters/python/hooks/setuptools-check-hook.sh

Reply:
@eadwu Let's get straight to the issue. The `checkPhase` definition that you introduce here has no effect, it's a dead code as far as I can tell. For example, if I do a simple modification:
```diff
diff --git a/pkgs/applications/version-management/sourcehut/todo.nix b/pkgs/applications/version-management/sourcehut/todo.nix
index b4928a7de5c..f172e8da33a 100644
--- a/pkgs/applications/version-management/sourcehut/todo.nix
+++ b/pkgs/applications/version-management/sourcehut/todo.nix
@@ -55,6 +55,7 @@ buildPythonPackage rec {
       origin=http://meta.sr.ht.local
     '';
   in ''
+false
     cp -f ${config} config.ini
     pytest tests/
   '';

```
the build will still succeed. I believe you either need to use `installCheckPhase` and then remove `doCheck = false` or just not commit this chunk of dead code.